### DISCUSSION TITLE
Subset bulk download?

### DISCUSSION THREAD

**Discussion by yakomaxa** [facebookresearch/esm#433] [ESM Metagenomic Atlas]
I have a list of MGnify ids and I would like to download all of the structure models corresponding to the listed ids from ESM-Atlas.
The size of the list is ~ tens of thousands. I guess a naive solution to download the structures is to compose a script using `for` and `wget`, but I think this is not so elegant. 

Are there better ways to perform bulk download for specific subsets from ESM-atlas?

**Comment by yakomaxa** [comment]
Probably, I noticed that https://github.com/facebookresearch/esm/issues/375 is a very similar issue.

**Reply by tomsercu** [reply]
Yes wget or curl or aria2c with a list of 10,000 urls to fetch is fine. Please comment in the issue if it works or you still see rate limiting errors.

For use cases where you want to access to the bulk dataset, please download the tarballs we provide here: https://github.com/facebookresearch/esm/blob/main/scripts/atlas/README.md

**Reply by yakomaxa** [reply]
In my experiences, too many programmatic access against public (traditional) biological databases sometimes results in my IP address being banned, but ESM Atlas seems to be exceptional case. Thank you very much for large server capacity of ESM Atlas.

**Reply by yakomaxa** [reply]
At this moment, `wget` with `for` iteration is working fine for me. It may take days, but I'm not in hurry and the procedure is enough for me. I'll read the discussion at https://github.com/facebookresearch/esm/issues/375 when I want to download  larger sets more rapidly.

**Reply by tomsercu** [reply]
Yes we have a 100rps rate limit now, and if you exceed you won't get "banned", just temporarily blocked.
You should look at the `aria2c` instructions on the page above, will be orders of magnitude faster than looping wget. At least you could loop `wget url &` so you send all the processes to background :)

**Reply by yakomaxa** [reply]
Thank you very much for detailed advice. I had refrained from running download script in parallel because of the fear of being "banned", but now I understand it would not happen to me as long as fetching from ESM Atlas. At the next step of my research, I will prepare 100x larger lists of MGnify ids so I should learn how to use aria2c with my lists instead of looping with wget as you suggested.
